src/client.js

```diff
@@ -716,20 +737,26 @@ class Client {
             that.application = new Application(that);
             return Promise.resolve();
         }
-        else if (credentials._type == "UserPassword") {
-            const user = credentials._getUser();
-            const password = credentials._getPassword();
-
-            const soapCall = this._prepareSoapCall("xtk:session", "Logon");
-            soapCall.writeString("login", user);
-            soapCall.writeString("password", password);
-            var parameters = null;
-            if (this._connectionParameters._options.rememberMe) {
-                parameters = soapCall.createElement("parameters");
-                parameters.setAttribute("rememberMe", "true");
+        else if (credentials._type == "UserPassword" || credentials._type == "BearerToken") {
```

Comment:
Not sure if the interest of the refactoring here. There’s only one or 2 lines in common between the 2 code branches, and I find the ternery operator and nested if statements confusing. Can you keep the user password case unchanged and add another if for the bearer token case and see what it looks like?